precompile/contract.go

```diff
@@ -15,12 +17,19 @@ const (
 	selectorLen = 4
 )
 
+// ContractRef is a reference to the contract's backing object
+type ContractRef interface {
+	Address() common.Address
+}
```

Comment:
Can we move this back into its original location and add an extra `CallFromPrecompile` function which will take in a normal `common.Address` type instead of `ContractRef` to reduce the amount of EVM code that needs to be changed?